deployment/config.go

```diff
@@ -179,6 +183,50 @@ type ExternalBucketSettings struct {
 	AmazonS3SSE             bool   `default:"false"`
 }
 
+// ExternalAuthProviderSettings contains the necessary data
+// to configure an external auth provider.
+type ExternalAuthProviderSettings struct {
+	// InstanceCount is the number of keycloak instances to deploy. It should be 0 or 1.
+	// If set to 0, the deployment will not include a keycloak instance.
+	InstanceCount int `default:"0" validate:"range:[0,1]"`
```

Comment:
Should we use an `Enable bool` instead of `InstanceCount int` if we're only allowing one instance?